l10n_ch_invoice_reports/models/mail_template.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2021 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+import base64
+
+from odoo import _, models
+
+
+class MailTemplate(models.Model):
+    _inherit = "mail.template"
+
+    def generate_email(self, res_ids, fields=None):
+        # Method overridden in order remove QR/ISR payslips and use
+        # Invoice report with payslip generated by this module
+
+        result = super().generate_email(res_ids, fields)
+        if self._should_print_invoice_attachment():
+            res_ids_to_templates = self.get_email_template(res_ids)
+            for res_id in self.env["account.move"].browse(res_ids):
+                related_record = self.env[self.model_id.model].browse(res_id.id)
+                template = res_ids_to_templates[res_id.id]
+                invoice_attachments = self.generate_invoice_attachment(
+                    template, res_id.id, related_record
+                )
+                result[res_id.id]["attachments"] = invoice_attachments
+        return result
+
+    def _should_print_invoice_attachment(self):
+        if (
+            self.model_id.model == "account.move"
+            and self.report_template.report_name
+            == "l10n_ch_invoice_reports.account_move_payment_report"
```

Comment:
the report name is not mandatory here, user can replace it on any report he wants to have

Reply:
the issue of email generation is that implementation in odoo doesn't rely on the report attached to template, the only thing we need here is to check that report is there, which one - doesn't matter 
non-blocking thou

Reply:
Please add a comment regarding this.